pkg/common/cluster/clusterutil.go

```diff
@@ -140,6 +142,38 @@ func WaitForClusterReadyPostScale(clusterID string, logger *log.Logger) error {
 	return waitForClusterReadyWithOverrideAndExpectedNumberOfNodes(clusterID, logger, false, false)
 }
 
+// WaitForClusterReadyPostWake blocks until the cluster is ready for testing, deletes errored pods, and then uses
+// healthcheck mechanisms appropriate for after the cluster has been scaled.
+func WaitForClusterReadyPostWake(clusterID string, logger *log.Logger) error {
+	log.Printf("Cluster %s just woke up, waiting for 10 minutes...", clusterID)
+	time.Sleep(10 * time.Minute)
+
+	restConfig, _, err := ClusterConfig(clusterID)
+	if err != nil {
+		return fmt.Errorf("Error getting cluster config: %v\n", err)
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return fmt.Errorf("Error generating Kube Clientset: %v\n", err)
+	}
+
+	list, err := kubeClient.CoreV1().Pods("").List(context.TODO(), v1.ListOptions{})
```

Comment:
I would expect that we might need to handle pagination here, as there could be a great many pods. Am I missing something there?

Reply:
Good call!